plugin/pkg/admission/persistentvolume/label/admission.go

```diff
@@ -111,6 +164,51 @@ func (l *persistentVolumeLabel) Admit(a admission.Attributes) (err error) {
 	return nil
 }
 
+func (l *persistentVolumeLabel) isPVCOkToDelete(pvc *api.PersistentVolumeClaim) bool {
+	// prevent deletion of a pvc that is bound to a pv
+	// if the pv is used by an active pod
+	// issue #45143
+
+	glog.Infof("IDC isPVCOkToDelete entered")
+
+	//check if pvc is bound to pv and associated pv name exists
+	if pvc != nil && pvc.Status.Phase == api.ClaimBound && pvc.Spec.VolumeName != "" {
+		glog.Infof("IDC isPVCOkToDelete pvc is bound")
+		//find the associated pv and check
+		volume, volumeErr := l.client.Core().PersistentVolumes().Get(pvc.Spec.VolumeName, metav1.GetOptions{})
+		if volumeErr == nil {
+			glog.Infof("IDC isPVCOkToDelete found bound volume %s", volume.Name)
+			//check all running and pending pods for this volume
+			pods, podErr := l.podLister.Pods(pvc.Namespace).List(labels.Everything())
+			if podErr == nil {
+				glog.Infof("IDC isPVCOkToDelete got pods")
+				for _, pod := range pods {
+					glog.Infof("IDC isPVCOkToDelete checking pod %q", pod.Name)
+					if pod.Status.Phase == api.PodRunning || pod.Status.Phase == api.PodPending {
```

Comment:
Since this is user triggered action, i think we will be okay with deleting PVC for terminated or completed pods.

I had another question - are we sure we don't want to allow people from deleting PVCs if pod is pending? a bound PVC is a real resource for which someone is paying money, whereas a pending pod is just an API object.  wondering, if people rely on this behaviour. Say - if user creates a pod and pod gets stuck in pending for whatever reason (no node available for scheduling may be), then user can't delete pvc without deleting pod. If he wrote some tooling/script that deleted PVCs, it would break now.